functions/Get-DbaRegisteredServer.ps1

```diff
@@ -0,0 +1,159 @@
+
+		.EXAMPLE
+			Get-DbaRegisteredServer -SqlInstance sqlserver2014a -Group HR\Development
+
+			Returns a list of servers in the HR and sub-group Development from the CMS on sqlserver2014a
+	#>
+	[CmdletBinding(DefaultParameterSetName = "Default")]
+	param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[PSCredential]$SqlCredential,
+		[Alias("Groups")]
+		[object[]]$Group,
+		[object[]]$ExcludeGroup,
+		[switch]$ExcludeCmsServer,
+		[switch]$Silent
+	)
+	begin {
+		function Find-CmsGroup {
+			[OutputType([object[]])]
+			[cmdletbinding()]
+			param(
+				$CmsGrp,
+				$Base = $null,
+				$Stopat
+			)
+			$results = @()
+			foreach ($el in $CmsGrp) {
+				if ($null -eq $Base -or [string]::IsNullOrWhiteSpace($Base) ) {
+					$partial = $el.name
+				}
+				else {
+					$partial = "$Base\$($el.name)"
+				}
+				if ($partial -eq $Stopat) {
+					return $el
+				}
+				else {
+					foreach ($elg in $el.ServerGroups) {
+						$results += Find-CmsGroup -CmsGrp $elg -Base $partial -Stopat $Stopat
+					}
+				}
+			}
+			return $results
+		}
+	}
+
+	process {
+		if (Test-FunctionInterrupt) { 
+			return 
+		}
+
+		$servers = @()
+		foreach ($instance in $SqlInstance) {
+			try {
+				$cmsStore = Get-DbaRegisteredServersStore -SqlInstance $instance -SqlCredential $SqlCredential -Silent:$Silent
+			}
+			catch {
+				Stop-Function -Message "Cannot access Central Management Server '$instance'." -ErrorRecord $_ -Continue
+				return
+			}
+
+			if (Test-Bound -ParameterName ExcludeGroup) {
+				$Group = ($cmsStore.DatabaseEngineServerGroup.ServerGroups | Where-Object Name -notin $ExcludeGroup).Name
+			}
+
+			if ($Group) {
+				foreach ($currentGroup in $Group) {
+					$cms = Find-CmsGroup -CmsGrp $cmsStore.DatabaseEngineServerGroup.ServerGroups -Stopat $currentGroup
+					if ($null -eq $cms) {
+						Write-Message -Level Output -Message "No groups found matching that name on instance '$instance'."
+						continue
+					}
+					$servers += ($cms.GetDescendantRegisteredServers())
+				}
+			}
+			else {
+				$cms = $cmsStore.DatabaseEngineServerGroup
+				$servers += ($cms.GetDescendantRegisteredServers())
+			}
+
+			if ($ExcludeCmsServer) {
+				$servers = ($servers | Where-Object { $_.ServerName -ne $instance})
+			}
+		}
+
+		foreach ($server in $servers) {
+			try {
+				$lookup = Resolve-DbaNetworkName $server.ServerName -Turbo -Silent
+				Add-Member -Force -InputObject $server -MemberType NoteProperty -Name ComputerName -Value $lookup.ComputerName
+				Add-Member -Force -InputObject $server -MemberType NoteProperty -Name IPAddress -Value $lookup.IPAddress -PassThru
+			}
+			catch {
+				Add-Member -Force -InputObject $server -MemberType NoteProperty -Name ComputerName -Value $null
+				Add-Member -Force -InputObject $server -MemberType NoteProperty -Name IPAddress -Value $null -PassThru
+			}
+		}
+	}
+}
```

Comment:
Please add the deprecation catch in an end {}

`Test-DbaDeprecation -DeprecatedOn "1.0.0" -Alias Test-DbaRegisteredServerName`
`Test-DbaDeprecation -DeprecatedOn "1.0.0"  -Alias Get-SqlRegisteredServerName`

@FriedrichWeinmann - is that how it works for mulitple or is it `-Alias Test-DbaRegisteredServerName, Get-SqlRegisteredServerName`

Reply:
I wasn't sure if that was needed in this one, since it's a new command (I didn't create an alias for it). I can add it if you let me know what is needed.

Reply:
You are right, i'll need to add it to psm1 and psd1 but yes, please make two lines (just confirmed with fred)